### PR TITLE
feat: implement persisted country selection; deps for writing tests

### PR DESCRIPTION
**Pull request by piperchester**
This implements #28. I use `window.sessionStorage` to persist the selected country as well as React's state to trigger a client-side refresh when the user selects and clears the option.

selected-country

Open to any suggestions, big or small on any of this!

## Also
- Added first `.test.js` file, `__snapshots__` and necessary Jest deps
- Added `selected-item` class to scope the white color for the country
- Added PR template
- Fixed console warnings in `RecentLinkItem`

## Todo
- [x] Tests
- [x] Clean up
- [x] Do not default to a country

### CODE REVIEW

**Comment by piperchester** [comment]
Hey Phil and Mike! Requested you as reviewers as I'm not sure who the best folks to reach out to are. Happy to elaborate on any of this!

File: src/components/LanguageDropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+     * refreshing, or returning to the site. We also bind the
+     * update to React's state, so that when a user re-selects
+     * an option, we trigger a state change to refresh the site.
+     *
+     * We use the ISO 639-1 language codes to map to specified languages, e.g.
+     * "EN" maps to "English".
+     */
+    const updateLanguage = (language) => {
+      setLanguage(language);
+      sessionStorage.setItem('lang_pref', language);
```
Comment (piperchester) [inline src/components/LanguageDropdown.jsx:25]: Went ahead with "lang_pref" as the key. WDYT?

File: src/components/LanguageDropdown.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import LanguageDropdown from "./LanguageDropdown";
+
+describe("LanguageDropdown", () => {
+  it("renders correctly", () => {
+    const tree = renderer
+      .create(<LanguageDropdown />)
+      .toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+});
```
Comment (piperchester) [inline src/components/LanguageDropdown.test.js:13]: First test! Just a simple sanity snapshot.

File: src/components/RecentLinkItem.jsx [hunk 3]
```diff
@@ -20,12 +20,13 @@ export default ({ link }) => {
...
               <Badge
+                key={tag}
                 variant="secondary"
                 className="ml-4"
                 style={{
                   marginLeft: '2em',
-                  'font-size': '60%',
-                  'font-weight': 300,
+                  'fontSize': '60%',
+                  'fontWeight': 300,
```
Comment (piperchester) [inline src/components/RecentLinkItem.jsx:29]: These fix 3 console warnings.

**Comment by philsturgeon** [comment]
Thanks for taking a swing at this! 

Sorry for any confusion, but this is only meant to be a country selector. Everyone is still stuck seeing english, it's just going to display content for Brazil or Australia or Bolivia, it's all about restricting to content relevant to you. A basic filter, which is persisted. 

Localiztion is out of scope, as #14 will give people language. If somebody else wants to view the site in French when they live in Mongolia that should be fine. 

Do you think you could tweak that to make it clear?

**Comment by piperchester** [comment]
> Do you think you could tweak that to make it clear?

For sure! So is the intent to fold the functionality from the "Filter site for your country" bit into a single dropdown? 

Also good timing I think - guessing the site's undergoing some maintenance but getting a "website is pending domain owner verification" on https://protect.earth/.

**Comment by philsturgeon** [comment]
Right, the goal was to make the "Filter site for your country" work properly.

1. persist the country of choice
2. display the country of choice
3. make sure it showed things from your country _and_ things which had no country, because a website available anywhere is a good thing to show to somebody wherever they are

Does that make sense? 

The hosted URL is https://actions.protect.earth whilst we switch the main page over to Squarespace, which is surprisingly slow at making SSL certificates. 😓

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,8 +1,6 @@
 {
   "editor.formatOnSave": true,
   "editor.formatOnPaste": true,
-  "cSpell.ignoreWords": [
-    "ecotricity",
-    "sunmills"
-  ]
-}
\ No newline at end of file
+  "editor.defaultFormatter": "esbenp.prettier-vscode",
```
Comment (piperchester) [inline .vscode/settings.json:4]: Added this to automatically invoke prettier on save. This streamlines my DX but can definitely pull this out if undesired!

File: src/components/CountrySelection.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * If the user has a country preference in session storage, we
+ * update the button text to "Change country".
+ */
+const CountrySelection = ({}) => {
+  // Get the country code. e.g., "BG" for "Bulgaria"
+  const [country] = useState(sessionStorage.getItem('country_pref'));
+
+  // Default to Great Britain
+  const countryName = Countries.fromAlpha2Code(country || 'GB').name;
+  const countryEmoji = Countries.fromAlpha2Code(country || 'GB').emoji;
```
Comment (piperchester) [inline src/components/CountrySelection.jsx:19]: Always look into the session storage first. Defaults to 'GB' which I could see being controversial. @ reviewers, wdyt?

File: src/pages/select-your-country.jsx [hunk 6]
```diff
@@ -1,14 +1,35 @@
...
+   * that persists through reloads. This enables users to have
+   * continuity when refreshing, or returning to the site.
+   *
+   * We also bind the update to React's state, so that when a user re-selects
+   * an option, we trigger a state change to refresh the site.
+   */
+  const updateCountry = (country) => {
+    setCountry(country);
+    sessionStorage.setItem('country_pref', country);
+  };
```
Comment (piperchester) [inline src/pages/select-your-country.jsx:31]: This is the core piece where we write to session storage, and update React's state.

**Comment by piperchester** [comment]
Took another stab at this @philsturgeon! Happy to make any desired changes.

File: src/components/CountrySelection.jsx (hunk 5 above)
Comment (philsturgeon) [reply src/components/CountrySelection.jsx:19]: Cn we make the default be no country? The idea of a filter is that it reduces the overall data to a subset of more relevant information, but that is an optional feature which is only used when folks need it.

Comment (piperchester) [reply src/components/CountrySelection.jsx:19]: Sure thing! Will fix this 👍

**Comment by piperchester** [comment]
Getting a chance to revisit this @philsturgeon - apologies for the delays! 

Following the contrib guidelines, do you mind sharing env variables for the new Airtable setup?

**Comment by philsturgeon** [comment]
Sorry or the delays here too! Agh.

This project is finally getting its new home, merged with climatechoice over here https://github.com/impactmakers/climatechoice/pull/161

The airtable source and directory approach is simply being moved over there, so this will still need to be done, just once the dust has settled of that PR. Could you subscribe to the PR and when its done try and help with a country selector?

**Comment by philsturgeon** [comment]
Btw I'll try and get the read-only version of the API up so people dont need EVN vars soon.

**Pull request closed without merging**